nifi-nar-bundles/nifi-adx-bundle/nifi-adx-processors/src/main/java/org/apache/nifi/processors/adx/AzureAdxIngestProcessor.java

```diff
@@ -0,0 +1,437 @@
+                case "TSV": ingestionProperties.setDataFormat(IngestionProperties.DataFormat.TSV); break;
+                case "TSVE": ingestionProperties.setDataFormat(IngestionProperties.DataFormat.TSVE); break;
+                case "TXT": ingestionProperties.setDataFormat(IngestionProperties.DataFormat.TXT); break;
+            }
+
+            switch(context.getProperty(IR_LEVEL).getValue()) {
+                case "IRL_NONE": ingestionProperties.setReportLevel(IngestionProperties.IngestionReportLevel.NONE); break;
+                case "IRL_FAIL": ingestionProperties.setReportLevel(IngestionProperties.IngestionReportLevel.FAILURES_ONLY); break;
+                case "IRL_FAS": ingestionProperties.setReportLevel(IngestionProperties.IngestionReportLevel.FAILURES_AND_SUCCESSES); break;
+            }
+
+            switch (context.getProperty(IR_METHOD).getValue()) {
+                case "IRM_TABLE": ingestionProperties.setReportMethod(IngestionProperties.IngestionReportMethod.TABLE); break;
+                case "IRM_QUEUE": ingestionProperties.setReportMethod(IngestionProperties.IngestionReportMethod.QUEUE); break;
+                case "IRM_TABLEANDQUEUE": ingestionProperties.setReportMethod(IngestionProperties.IngestionReportMethod.QUEUE_AND_TABLE); break;
+            }
+
+            if (StringUtils.equalsIgnoreCase(context.getProperty(FLUSH_IMMEDIATE).getValue(),"true")) {
+                ingestionProperties.setFlushImmediately(true);
+            } else {
+                ingestionProperties.setFlushImmediately(false);
+            }
+
+            getLogger().info("Ingesting with: "
+                    + "dataFormat - "+ ingestionProperties.getDataFormat() + "|"
+                    + "ingestionMapping - " + ingestionProperties.getIngestionMapping().getIngestionMappingReference() + "|"
+                    + "reportLevel - " + ingestionProperties.getReportLevel() + "|"
+                    + "reportMethod - " + ingestionProperties.getReportMethod() + "|"
+                    + "databaseName - " + ingestionProperties.getDatabaseName() + "|"
+                    + "tableName - " + ingestionProperties.getTableName() + "|"
+                    + "flushImmediately - " + ingestionProperties.getFlushImmediately()
+            );
+
+            StreamSourceInfo info = new StreamSourceInfo(in);
+
+            IngestionResult result = client.ingestFromStream(info, ingestionProperties);
+
+            List<IngestionStatus> statuses = result.getIngestionStatusCollection();
+
+            if(StringUtils.equalsIgnoreCase(context.getProperty(WAIT_FOR_STATUS).getValue(),"ST_SUCCESS")) {
+                while (statuses.get(0).status == OperationStatus.Pending) {
+                    Thread.sleep(50);
+                    statuses = result.getIngestionStatusCollection();
+                    if(statuses.get(0).status == OperationStatus.Succeeded) {
```

Comment:
Has any end to end test been done on it  ? Configuring it from the UI in NIFI and run it end to end ? Let us document this ?